src/get_programming_with_js_next/module_7_exercise.es6

```diff
@@ -0,0 +1,45 @@
+/*
+  Write a function that you can use to tag template literals that checks each
+  value if it is an object, and if so, converts the object into a string of
+  key/value pairs before interpolation.
+
+  Input:
+
+  const props = {
+    src: 'http://fillmurray.com/100/100',
+    alt: 'Bill Murray'
+  };
+
+  const img = withProps`<img ${props}>`;
+
+  Output:
+
+  <img src="http://fillmurray.com/100/100" alt="Bill Murray">
+*/
+
+const PROPS = {
+  src: 'http://fillmurray.com/100/100',
+  alt: 'Bill Murray'
+};
+
+function formatAttributes(attributes) {
+  let formattedAttributes = '';
+
+  for (let attribute in attributes) {
+    if ({}.hasOwnProperty.call(attributes, attribute)) {
+      formattedAttributes += `${attribute}="${attributes[attribute]}" `;
+    }
+  }
+
+  return formattedAttributes.trim();
+}
+
+function withProps(strParts, props) {
```

Comment:
Nit - One more neat trick that I didn't notice yesterday, not necessary, just fancy:
```javascript
function withProps([openTag, closeTag], props) {
  return `${openTag}${formatAttributes(props)}${closeTag}`;
}
```

You can destructure the array in the param itself. Not a necessity, both are legible :)

Reply:
👍